### PR TITLE
feat(auth-bundler): implementation

### PR DESCRIPTION
**Pull request by CptSchnitz**

### CODE REVIEW

**Comment by CptSchnitz** [comment]
@syncush so you have a new notification.

File: packages/auth-bundler/tests/utils/client.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { faker } from '@faker-js/faker';
+import { IClient } from '@map-colonies/auth-core';
+
+export function getFakeClient(includeGeneratedFields: boolean): IClient {
+  const firstName = faker.name.firstName();
+  const lastName = faker.name.lastName();
+  const client: IClient = {
+    name: faker.internet.userName(firstName, lastName),
```
Comment (syncush) [inline packages/auth-bundler/tests/utils/client.ts:8]: what about names with special characters ? we should also include & % _ - * $ and such.
We already have a client name with & in its name

**Review by syncush — CHANGES_REQUESTED** [review]
You also have in `[packages/auth-core/src/model/key.ts](https://github.com/MapColonies/opa-la/pull/10/files#diff-0e56735564612c097c4090c33b630ae6f73a21d9c9faf0fd3f772afa7f801995)` 
eslint warn on defined but never used.

File: packages/auth-bundler/tests/utils/client.ts (hunk 1 above)
Comment (CptSchnitz) [reply packages/auth-bundler/tests/utils/client.ts:8]: the file was not used in the tests so i removed it.

File: packages/auth-bundler/tests/utils/key.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { faker } from '@faker-js/faker';
+import { JWKPrivateKey, JWKPublicKey } from '@map-colonies/auth-core';
+
+const THREE = 3;
```
[the remark this answers is not in the document]
Comment (CptSchnitz) [reply packages/auth-bundler/tests/utils/key.ts:4]: changed

File: packages/auth-bundler/tests/opa.spec.ts [hunk 3]
```diff
@@ -0,0 +1,143 @@
+import { tmpdir } from 'node:os';
+import path from 'node:path';
+import { existsSync } from 'node:fs';
```
[the remark this answers is not in the document]
Comment (CptSchnitz) [reply packages/auth-bundler/tests/opa.spec.ts:3]: removed

File: packages/auth-bundler/tests/db.spec.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+import { Asset, Connection, DbConfig, Environment, Key, createConnectionOptions } from '@map-colonies/auth-core';
+import { DataSource } from 'typeorm';
+import { BundleDatabase } from '../src/db';
+import { getMockKeys } from './utils/key';
+import { getFakeAsset } from './utils/asset';
+import { getFakeConnection } from './utils/connection';
+
+describe('db.ts', function () {
+  let dataSource: DataSource;
+  const asset = { ...getFakeAsset(), environment: [Environment.PRODUCTION], name: 'aviaviavi' };
```
[the remark this answers is not in the document]
Comment (CptSchnitz) [reply packages/auth-bundler/tests/db.spec.ts:13]: indeed

File: packages/auth-bundler/src/logger.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+/**
+ * A function to disable the logger in the package.
+ * @group logger
+ * @see {@link https://github.com/MapColonies/js-logger }
+ * @see {@link https://github.com/pinojs/pino }
+ */
+export function unsetLogger(): void {
+  logger = undefined;
```
[the remark this answers is not in the document]
Comment (CptSchnitz) [reply packages/auth-bundler/src/logger.ts:29]: removed after talking with you

File: packages/auth-bundler/src/db.ts [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+    const promises = await Promise.all([assets, connections, key]);
+
+    return { assets: promises[0], connections: promises[1], key: promises[2], environment: versions.environment };
+  }
+
+  private async getAssetsVersions(environment: string): Promise<{ name: string; version: number }[]> {
+    const subQuery = this.assetRepository
+      .createQueryBuilder('asset')
+      .select(['name', 'MAX(version)'])
+      // .where(':environment = ANY (environment)', { environment })
```
[the remark this answers is not in the document]
Comment (CptSchnitz) [reply packages/auth-bundler/src/db.ts:104]: mistake

**Review by syncush — APPROVED** [review]

**Pull request merged**